e3sm_diags/driver/zonal_mean_xy_driver.py

```diff
@@ -45,9 +47,15 @@ def run_diag(parameter: CoreParameter) -> CoreParameter:
     variables = parameter.variables
     seasons = parameter.seasons
     ref_name = getattr(parameter, "ref_name", "")
-    # zonal mean xy only tested with global datasets
     regions = parameter.regions
 
+    for region in regions:
+        if region != "global":
+            raise RuntimeError(
+                f"Region ({region}) is not supported. Only global region is currently "
+                "supported for the zonal_mean_xy set."
+            )
```

Comment:
I moved this check out of `_calculate_zonal_mean()` to the top-level of `run_diags()`.